lib/engine-jit/src/serialize.rs

```diff
@@ -38,9 +32,61 @@ pub struct SerializableCompilation {
 
 /// Serializable struct that is able to serialize from and to
 /// a `JITArtifactInfo`.
-#[derive(Serialize, Deserialize, MemoryUsage)]
+#[derive(Serialize, Deserialize, MemoryUsage, Archive, RkyvDeserialize, RkyvSerialize)]
 pub struct SerializableModule {
     pub compilation: SerializableCompilation,
     pub compile_info: CompileModuleInfo,
     pub data_initializers: Box<[OwnedDataInitializer]>,
 }
+
+fn to_serialize_error(err: impl std::error::Error) -> SerializeError {
+    SerializeError::Generic(format!("{}", err))
+}
+
+impl SerializableModule {
+    pub fn serialize(&self) -> Result<Vec<u8>, SerializeError> {
+        let mut serializer = SharedSerializerAdapter::new(WriteSerializer::new(vec![]));
+        let pos = serializer
+            .serialize_value(self)
+            .map_err(to_serialize_error)? as u64;
+        let mut serialized_data = serializer.into_inner().into_inner();
+        serialized_data.extend_from_slice(&pos.to_le_bytes());
+        if cfg!(target_endian = "big") {
+            serialized_data.extend_from_slice(&[b'b']);
+        } else if cfg!(target_endian = "little") {
+            serialized_data.extend_from_slice(&[b'l']);
+        }
+        Ok(serialized_data)
+    }
+
+    pub unsafe fn deserialize(metadata_slice: &[u8]) -> Result<Self, DeserializeError> {
+        let archived = Self::archive_from_slice(metadata_slice)?;
+        Self::deserialize_from_archive(archived)
+    }
+
+    unsafe fn archive_from_slice<'a>(
```

Comment:
`unsafe` functions must have a `# Safety` section explaining how they can be used

Reply:
Only pub functions though :)

Reply:
I don't know about that. What the `unsafe` keyword means is "this function has invariants that cannot be checked by the compiler, you must manually ensure that they're true or the safety model of Rust breaks down".

An `unsafe` function without documentation explaining how it can be used is kind of nonsense (I suppose it means "you can't ever call this function"), even if it's not public.